test/src/specs/sync/chain_forks.rs

```diff
@@ -552,6 +553,63 @@ impl Spec for ForksContainSameTransactions {
     }
 }
 
+pub struct ForkedTransaction;
+
+impl Spec for ForkedTransaction {
+    crate::name!("forked_transaction");
+
+    crate::setup!(num_nodes: 2, connect_all: false);
+
+    // Case: Check TxStatus of transaction on main-fork, verified-fork and unverified-fork
+    fn run(&self, net: Net) {
+        let node0 = &net.nodes[0];
+        let node1 = &net.nodes[1];
+        let finalization_delay_length = node0.consensus().finalization_delay_length();
+
+        net.exit_ibd_mode();
+        let fixed_point = node0.get_tip_block_number();
+        let tx = node1.new_transaction_spend_tip_cellbase();
+
+        // node0 doesn't have `tx`      => TxStatus: None
+        // node1 have `tx` on main-fork => TxStatus: Some(Committed)
```

Comment:
the main-fork is the main-chain?

Reply:
Thanks for taking the time to seriously review the test-cases. I have added some code comments to explain the process of test-cases. Please take a look again.

And `assert_block` and `assert_chain` has been removed based on your suggestions.